### PR TITLE
[SPARK-28923][SQL] Deduplicate the codes 'multipartIdentifier' and 'identifierSeq'

### PR DESCRIPTION
**Pull request by xy-xin**
### What changes were proposed in this pull request?

In `sqlbase.g4`, `multipartIdentifier` and `identifierSeq` have the same functionality. We'd better deduplicate them.

### Why are the changes needed?
Deduplicate the codes which have the same function.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Existing tests.

### CODE REVIEW

**Comment by maropu** [comment]
ok to test

**Comment by maropu** [comment]
The change looks ok to me.

**Comment by maropu** [comment]
cc: @dongjoon-hyun @viirya

File: sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 [hunk 1]
```diff
@@ -567,11 +567,7 @@ sampleMethod
     ;
 
 identifierList
-    : '(' identifierSeq ')'
-    ;
-
-identifierSeq
-    : ident+=errorCapturingIdentifier (',' ident+=errorCapturingIdentifier)*
```
[the remark this answers is not in the document]
Comment (maropu) [reply sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: oh, right. I missed it...

[the remark this answers is not in the document]
Comment (xianyinxin) [reply sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: sorry, missed that.

**Pull request closed without merging**